src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -80,6 +80,25 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
         confirmation_message = "I confirm I have an eligible Windows Server license with Azure Hybrid Benefit to apply this benefit to AKS on HCI or Windows Server. Visit https://aka.ms/ahb-aks for details"
         utils.user_confirmation(confirmation_message, yes)
 
+    if least_privilege is True:
+        # pre-checks specific to least privilege scenario
+        # check if config settings are passed
+        if config_settings is None:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception="Configuration settings are not passed", fault_type=consts.Config_Settings_Not_Passed_Least_Privileges_Fault_Type, summary="Configuration settings are not passed which are mandatory when onboarding with leastPrivileges")
+            raise ValidationError("Configuration settings are not passed", "Please pass required configuration settings using '--config-settings' flag while onboarding with least-privilege enabled.")
+        else:
+            logger.warning("You are onboarding your k8s cluster to Azure Arc in least privileges mode. Please ensure you have met all the pre-requisites.")
```

Comment:
should we give a link to the pre-reqs here?

Reply:
we're not doing public preview currently so we don't have a public aka.ms link as of now

Reply:
We can add after we do public preview